apps/rush/src/utilities/PackageReviewChecker.ts

```diff
@@ -3,47 +3,59 @@
 
 import {
   IPackageJson,
+  ApprovedPackagesPolicy,
   RushConfiguration,
   RushConfigurationProject,
-  PackageReviewConfiguration,
   Utilities
 } from '@microsoft/rush-lib';
 
-export default class PackageReviewChecker {
-  private _rushConfiguration: RushConfiguration;
-  private _packageReviewConfiguration: PackageReviewConfiguration;
-
-  constructor(rushConfiguraton: RushConfiguration) {
-    this._rushConfiguration = rushConfiguraton;
-    this._packageReviewConfiguration = PackageReviewConfiguration.loadFromFile(rushConfiguraton.packageReviewFile);
-  }
+export class PackageReviewChecker {
```

Comment:
I have been attempting to separate the file format parsing from the business logic.  The "src/data" classes only deal with loading files from disk, which is already a very complicated process.

---
In reply to: [116289718](https://github.com/Microsoft/web-build-tools/pull/181#discussion_r116289718) [](ancestors = 116289718)